### DISCUSSION TITLE
RollWorkload with New Config

### DISCUSSION THREAD

**Discussion by twcchu** [vectordotdev/vector#21632] [Q&A]
Hello!

We are running vector aggregator as a helm chart with configuration through existingConfigMap. When we update the ConfigMap, it gets updated in the Vector pods, but the changes are not applied until the statefulset is restarted manually. We have `rollWorkload` set to the default `true`. I don't see the checksum annotation on the pods that should roll the workloads (if I am understanding that configuration correctly). Is there something I am missing that will allow the pods to restart automatically?

**Comment by twcchu** [comment]
There is a thread on the Discord Server from 2022 that says:
> the aggregator chart will redeploy the pods when the configmap holding the configuration changes

but that is not happening for my Vector deployment. Is the only other option to change the `rollWorkload` setting to `false` and add the `--watch-config` argument so that the config changes are applied?

**Reply by jszwedko** [reply]
There's a workaround in https://github.com/vectordotdev/helm-charts/issues/351 that may be applicable here too.

**Reply by jszwedko** [reply]
Though there it claims that configmap changes should be causing redeployments.
